### PR TITLE
Refactor MethodAction class

### PR DESCRIPTION
**Pull request by viman**

### CODE REVIEW

File: src/MethodAction.ts [hunk 1]
```diff
@@ -4,25 +4,15 @@ export class MethodAction {
...
-    constructor(public methodName: string, public args: Array<any>) {
+    constructor(public methodName: string, public args: any[]) {
         this.callIndex = ++MethodAction.globalCallIndex;
     }
 
     public isApplicable(methodName: string, matchers: Matcher[]): boolean {
-        const methodNameMatch = this.methodName === methodName;
-        const argumentsCountMatch = this.args.length === matchers.length;
-        if (!methodNameMatch || !argumentsCountMatch) {
+        if (this.methodName !== methodName || this.args.length !== matchers.length) {
```
Comment (NagRock) [inline src/MethodAction.ts]: Sorry but I think that named const are more readable than this.

File: src/MethodAction.ts [hunk 2]
```diff
@@ -4,25 +4,15 @@ export class MethodAction {
...
-        let allValid = true;
-        let index: number = 0;
-        for (const arg of this.args) {
-            if (matchers[index] && !matchers[index].match(arg)) {
-                allValid = false;
-            }
-            index++;
-        }
-        return allValid;
+        return matchers.every((matcher: Matcher, index: number) => matcher.match(this.args[index]));
```
Comment (NagRock) [inline src/MethodAction.ts]: Another one great change :)

File: src/MethodAction.ts (hunk 1 above)
Comment (viman) [reply src/MethodAction.ts]: You are right! Reverting that

**Comment by NagRock** [comment]
@viman thanks again for simplifying implementation! 🎉

**Pull request merged**